drivers/filesystems/fastfat/finfo.c

```diff
@@ -804,6 +802,9 @@ VfatSetRenameInformation(
     vfatSplitPathName(&NewName, &NewPath, &NewFile);
     DPRINT("New dir: %wZ, New file: %wZ\n", &NewPath, &NewFile);
 
+            if (IsDotOrDotDot(NewFile))
+                return STATUS_OBJECT_NAME_INVALID;
```

Comment:
You should `goto Cleanup` here, like other failure paths, to make sure we don't leak anything.

Reply:
Fixed. Thanks.